ignite/cmd/network_chain_show_accounts.go

```diff
@@ -22,13 +23,17 @@ func newNetworkChainShowAccounts() *cobra.Command {
 		RunE:  networkChainShowAccountsHandler,
 	}
 
+	c.Flags().AddFlagSet(flagSetAccountPrefixes())
+
 	return c
 }
 
 func networkChainShowAccountsHandler(cmd *cobra.Command, args []string) error {
 	session := cliui.New()
 	defer session.Cleanup()
 
+	addressPrefix := getAddressPrefix(cmd)
```

Comment:
Don't we already know address prefix from chain's genesis and gentx's on SPN?

Query commands may not necessarily build and init the chain locally so we may not able to access to genesis.

Gentxs can be retrieved from chain but we may not be able to know if they are valid (e.g. account prefix is same as chain's) before actually building, initializing and running a simulation.

Both ways doesn't look optimal right now, maybe we should add an account prefix argument/flag to the `chain publish` cmd? This is also a bit shallow as it requires one more info from the coordinator.  

[Here](https://github.com/ignite-hq/cli/blob/ef23320abc57633d0409d0826f4898c825d8a770/ignite/services/network/join.go#L101) we change the delegator address prefix to SPN, maybe we shouldn't, what was the reason for that? Maybe we can directly send the original account to SPN and use its prefix and trust that gentx has the proper account prefix. But later in the `chain launch` command we should validate that all the delegator address prexies are the same.

Any ideas how can we determine address prefix automatically? cc @lubtd

Reply:
> [Here](https://github.com/ignite-hq/cli/blob/ef23320abc57633d0409d0826f4898c825d8a770/ignite/services/network/join.go#L101) we change the delegator address prefix to SPN, maybe we shouldn't, what was the reason for that? Maybe we can directly send the original account to SPN and use its prefix and trust that gentx has the proper account prefix. But later in the chain launch command we should validate that all the delegator address prexies are the same for all gentxs.

The SPN does not accept addresses without the `spn` prefix. The `validateBasic` methods will fail. It will be hard without the genesis file. Maybe we can save the address prefix and the stake denom into the spn store for we can easily retrieve

Reply:
It is still a bit weird to ask the address prefix from user. I propose we try to find alternatives before making a conclusion.

Reply:
We use `spn` prefix in `spn` for everything because as @Pantani mentions this is the global configuration on `spn` and `spn` prefix must be used to use `validateBasic `
But another reason is also for flexibility. The address prefix of a chain is an information available in the source code of the chain, in `sdk` configs. If the user would change the config in the source code of the chain then all on-chain stored address prefix would need to be changed, which is not ideal. It's cleaner to store everything with `spn` and construct the genesis with the correct prefix once the chain is ready to be launched and the prefix defined.

> Query commands may not necessarily build and init the chain locally so we may not able to access to genesis.

The prefix is an information in the source of the chain so this would be a requirement to automatically detect the prefix

My suggestion is to show the accounts with `spn` prefix by default, as stored on `spn`, and provide a flag to show the address with a custom prefix, like the `ignite account list` command